clients/unshaded/src/main/java/tachyon/client/file/TachyonFileSystemCore.java

```diff
@@ -166,6 +167,15 @@ TachyonFile openIfExists(TachyonURI path, OpenOptions options) throws IOExceptio
       TachyonException;
 
   /**
+   * Persists a {@link TachyonFile} to the UnderFileSystem.
+   *
+   * @param file the TachyonFile to persist
+   * @return true if the persist succeeded, false otherwise
+   */
+  boolean persistFile(TachyonFile file, PersistOptions options)
```

Comment:
I understand ... the fact that the function takes TachyonFile as an argument should be sufficient to indicate that this only works for files ... unfortunately, TachyonFile can also be a directory =_=

what happens if you pass a TachyonFile that is a directory? on a related note, why not allow persisting directories? and what about recursive option?